.azure-pipelines.yaml

```diff
@@ -45,30 +28,66 @@ jobs:
       codeCoverageTool: 'cobertura'
       summaryFileLocation: '$(System.DefaultWorkingDirectory)/**/coverage.cobertura.xml'  
 
-  - task: alanwales.resharper-code-analysis.custom-build-task.ResharperCli@1
-    displayName: 'Automated code quality checks'
-    inputs:
-      SolutionOrProjectPath: 'kubernetes-client.sln'
-
-      FailBuildOnCodeIssues: false
-
-    continueOnError: true
-
   - task: DotNetCoreCLI@2
     displayName: 'dotnet pack'
     inputs:
       command: pack
       packagesToPack: src/KubernetesClient/KubernetesClient.csproj
       packDirectory: '$(Build.ArtifactStagingDirectory)/nupkg'
-      majorVersion: 1
-      minorVersion: 4      
-      versioningScheme: byPrereleaseNumber

```

Comment:
why versioning was removed?
it is to replace git verison

Reply:
The version numbers are embedded in the assemblies and NuGet packages you create. If you use build numbers which depend on external factors (such as the date and time the package was built or the CI job number), your build output (the binaries) will be different every time you build, _even if you build from the same source_.

This breaks [reproducible builds](https://reproducible-builds.org/). Having reproducible builds means I can clone this repository, build locally, get the SHA hash of the NuGet package it builds, and make sure it's the same which is uploaded on NuGet. It helps enhance trust.

The Kubernetes project is working towards reproducible builds - see https://github.com/kubernetes/kubernetes/issues/70131, and I think it would be good if the same holds true for the .NET client.

Reply:
IMHO, I prefer to `each build is an individual`.  
The build machine cannot be exactly the same.
For example, when your update dotnet 1 to dotnet 2, the output/nuget of compiler might be changed even the source codes are totally the same.

The nuget is a snapshot and the release team is responsible for the `official release`.

Maybe something like docker image could make the `build reproducible`.

@brendandburns what is your opinion?